File: tradeit-android-sdk/src/main/java/it/trade/android/sdk/model/TradeItPreviewStockOrEtfOrderResponseParcelable.java
package it.trade.android.sdk.model;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;
import java.util.List;

import it.trade.model.reponse.OrderDetails;
import it.trade.model.reponse.TradeItPreviewStockOrEtfOrderResponse;
import it.trade.model.reponse.Warning;

public class TradeItPreviewStockOrEtfOrderResponseParcelable implements Parcelable {

    protected String orderId;
    protected List<String> ackWarningsList;
    protected List<String> warningsList;
    protected TradeItOrderDetailsParcelable orderDetails;

    TradeItPreviewStockOrEtfOrderResponseParcelable(TradeItPreviewStockOrEtfOrderResponse response) {
        this.orderId = response.orderId;
        this.ackWarningsList = response.ackWarningsList;
        this.warningsList = response.warningsList;
        this.orderDetails = new TradeItOrderDetailsParcelable(response.orderDetails);
    }

    @Override
    public String toString() {
        return "TradeItPreviewStockOrEtfOrderResponseParcelable{" +
                "orderId='" + orderId + '\'' +
                ", orderDetails=" + orderDetails +
                '}';
    }

    public String getOrderId() {
        return orderId;
    }

    /**
     *
     * @deprecated Use orderDetails.warnings
     */
    @Deprecated
    public List<String> getAckWarningsList() {
        return ackWarningsList;
    }

    /**
     *
     * @deprecated Use orderDetails.warnings
     */
    @Deprecated
    public List<String> getWarningsList() {
        return warningsList;
    }

    public TradeItOrderDetailsParcelable getOrderDetails() {
        return orderDetails;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeStringList(this.ackWarningsList);
        dest.writeParcelable(this.orderDetails, flags);
        dest.writeString(this.orderId);
        dest.writeStringList(this.warningsList);
    }

    public TradeItPreviewStockOrEtfOrderResponseParcelable() {
    }

    protected TradeItPreviewStockOrEtfOrderResponseParcelable(Parcel in) {
        this.ackWarningsList = in.createStringArrayList();
        this.orderDetails = in.readParcelable(OrderDetails.class.getClassLoader());
        this.orderId = in.readString();
        this.warningsList = in.createStringArrayList();
    }

    public static final Parcelable.Creator<TradeItPreviewStockOrEtfOrderResponseParcelable> CREATOR = new Parcelable.Creator<TradeItPreviewStockOrEtfOrderResponseParcelable>() {
        @Override
        public TradeItPreviewStockOrEtfOrderResponseParcelable createFromParcel(Parcel source) {
            return new TradeItPreviewStockOrEtfOrderResponseParcelable(source);
        }

        @Override
        public TradeItPreviewStockOrEtfOrderResponseParcelable[] newArray(int size) {
            return new TradeItPreviewStockOrEtfOrderResponseParcelable[size];
        }
    };
}

File: tradeit-android-sdk/src/main/kotlin/it/trade/android/sdk/model/TradeItPreviewStockOrEtfOrderResponseParcelable.kt
package it.trade.android.sdk.model

import android.os.Parcel
import android.os.Parcelable
import it.trade.model.reponse.OrderDetails
import it.trade.model.reponse.TradeItPreviewStockOrEtfOrderResponse

class TradeItPreviewStockOrEtfOrderResponseParcelable : Parcelable {

    var orderId: String = ""

    @get:Deprecated("Use orderDetails.warnings")
    var ackWarningsList: List<String> = arrayListOf()

    @get:Deprecated("Use orderDetails.warnings")
    var warningsList: List<String> = arrayListOf()
    var orderDetails: TradeItOrderDetailsParcelable? = null

    internal constructor(response: TradeItPreviewStockOrEtfOrderResponse) {
        this.orderId = response.orderId
        this.ackWarningsList = response.ackWarningsList
        this.warningsList = response.warningsList
        this.orderDetails = TradeItOrderDetailsParcelable(response.orderDetails)
    }

    override fun toString(): String {
        return "TradeItPreviewStockOrEtfOrderResponseParcelable{" +
                "orderId='" + orderId + '\''.toString() +
                ", orderDetails=" + orderDetails +
                '}'.toString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeStringList(this.ackWarningsList)
        dest.writeParcelable(this.orderDetails, flags)
        dest.writeString(this.orderId)
        dest.writeStringList(this.warningsList)
    }

    constructor() {}

    protected constructor(`in`: Parcel) {
        this.ackWarningsList = `in`.createStringArrayList()
        this.orderDetails = `in`.readParcelable(OrderDetails::class.java.getClassLoader())
        this.orderId = `in`.readString()
        this.warningsList = `in`.createStringArrayList()
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<TradeItPreviewStockOrEtfOrderResponseParcelable> = object : Parcelable.Creator<TradeItPreviewStockOrEtfOrderResponseParcelable> {
            override fun createFromParcel(source: Parcel): TradeItPreviewStockOrEtfOrderResponseParcelable {
                return TradeItPreviewStockOrEtfOrderResponseParcelable(source)
            }

            override fun newArray(size: Int): Array<TradeItPreviewStockOrEtfOrderResponseParcelable?> {
                return arrayOfNulls(size)
            }
        }
    }
}
